include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -8461,119 +8038,49 @@ FEFaceEvaluation<dim,
-        std::array<unsigned int, VectorizedArrayType::size()>
-          face_orientation_ = {};
-
-        cells_[0]            = this->cell;
-        face_no_[0]          = this->face_no;
-        face_orientation_[0] = this->face_orientation;
-
-        if (fe_degree > -1)
-          {
-            return internal::FEFaceEvaluationImplGatherEvaluateSelector<
-              dim,
-              Number,
-              VectorizedArrayType>::template run<fe_degree,
-                                                 n_q_points_1d>(
-              n_components,
-              1,
-              internal::get_beginning<Number>(input_vector),
-              shared_vector_data,
-              *this->data,
-              *this->dof_info,
-              this->begin_values(),
-              this->begin_gradients(),
-              this->begin_hessians(),
-              this->scratch_data.begin(),
-              evaluation_flag & EvaluationFlags::values,
-              evaluation_flag & EvaluationFlags::gradients,
-              evaluation_flag & EvaluationFlags::hessians,
-              this->active_fe_index,
-              this->first_selected_component,
-              cells_,
-              face_no_,
-              this->subface_index,
-              this->dof_access_index,
-              face_orientation_,
-              this->descriptor->face_orientations);
-          }
-        else
-          return internal::
-            FEFaceEvaluationFactory<dim, Number, VectorizedArrayType>::
-              gather_evaluate(n_components,
-                              1,
-                              internal::get_beginning<Number>(input_vector),
-                              shared_vector_data,
-                              *this->data,
-                              *this->dof_info,
-                              this->begin_values(),
-                              this->begin_gradients(),
-                              this->begin_hessians(),
-                              this->scratch_data.begin(),
-                              evaluation_flag & EvaluationFlags::values,
-                              evaluation_flag & EvaluationFlags::gradients,
-                              evaluation_flag & EvaluationFlags::hessians,
-                              this->active_fe_index,
-                              this->first_selected_component,
-                              cells_,
-                              face_no_,
-                              this->subface_index,
-                              this->dof_access_index,
-                              face_orientation_,
-                              this->descriptor->face_orientations);
-      }
-  };
-
-  if (!fu())
+          *this);
+      else
+        internal::FEFaceEvaluationGatherFactory<
+          dim,
+          Number,
+          VectorizedArrayType>::evaluate(n_components,
+                                         evaluation_flag,
```

Comment:
in many places you changed the name to `integration_flag`, but here it is still `evaluation_flag`. Intentionally?